src/services/db/__tests__/GitFileCommitService.spec.ts

```diff
@@ -0,0 +1,555 @@
+        })
+
+        // Assert
+        expect(createSpy).toHaveBeenCalledWith(
+          sessionData.siteName,
+          sessionData.isomerUserId,
+          content,
+          directoryName,
+          fileName,
+          isMedia ? "base64" : "utf-8",
+          STAGING_BRANCH
+        )
+        expect(pushSpy).toHaveBeenCalledWith(sessionData, expect.any(Boolean))
+        expect(result).toEqual({ sha: "new-sha" })
+      })
+
+      it("should create a file and commit only to staging branch when not whitelisted", async () => {
+        const content = "file content"
+        const fileName = "file.txt"
+        const directoryName = "directory"
+        const isMedia = false
+
+        const createSpy = jest
+          .spyOn(gitFileSystemService, "create")
+          .mockResolvedValue(okAsync({ newSha: "new-sha" }))
+        const pushSpy = jest
+          .spyOn(gitFileCommitService, "pushToGithub")
+          .mockResolvedValue()
+
+        jest
+          .spyOn(gbUtils, "isReduceBuildTimesWhitelistedRepo")
+          .mockReturnValue(false)
+
+        // Act
+        const result = await gitFileCommitService.create(sessionData, {
+          content,
+          fileName,
+          directoryName,
+          isMedia,
+        })
+
+        // Assert
+        expect(createSpy).toHaveBeenCalledWith(
+          sessionData.siteName,
+          sessionData.isomerUserId,
+          content,
+          directoryName,
+          fileName,
+          isMedia ? "base64" : "utf-8",
+          STAGING_BRANCH
+        )
+        expect(createSpy).toHaveBeenCalledTimes(1) // create is called only once for the staging branch
+        expect(pushSpy).toHaveBeenCalledWith(sessionData, false) // push is called with shouldUpdateStagingLite as false
+        expect(result).toEqual({ sha: "new-sha" })
+      })
+    })
+
+    describe("update", () => {
+      it("should update a file and push to GitHub", async () => {
+        // Arrange
+        const fileContent = "updated file content"
+        const sha = "old-sha"
+        const fileName = "file.txt"
+        const directoryName = "directory"
+
+        const updateSpy = jest
+          .spyOn(gitFileSystemService, "update")
+          .mockResolvedValue(okAsync("new-sha"))
+        const pushSpy = jest
+          .spyOn(gitFileCommitService, "pushToGithub")
+          .mockResolvedValue()
+
+        // Act
+        const result = await gitFileCommitService.update(sessionData, {
+          fileContent,
+          sha,
+          fileName,
+          directoryName,
+        })
+
+        // Assert
+        const filePath = directoryName
+          ? `${directoryName}/${fileName}`
+          : fileName
+        expect(updateSpy).toHaveBeenCalledWith(
+          sessionData.siteName,
+          filePath,
+          fileContent,
+          sha,
+          sessionData.isomerUserId,
+          STAGING_BRANCH
+        )
+        expect(pushSpy).toHaveBeenCalledWith(sessionData, expect.any(Boolean))
```

Comment:
we should check if push was done to both staging & lite branch right?

Reply:
applies to all the test cases :/ 
thanks for the catch, added [33bff68](https://github.com/isomerpages/isomercms-backend/pull/1047/commits/33bff68ae09ef4af6831fb861834da5693fd4d97)